src/server/search/search_family.cc

```diff
@@ -110,6 +117,47 @@ optional<SearchParams> ParseSearchParamsOrReply(CmdArgList args, ConnectionConte
       continue;
     }
 
+    // RETURN {num} [{ident} AS {name}...]
+    if (ArgS(args, i) == "RETURN") {
+      if (i + 1 >= args.size()) {
+        (*cntx)->SendError(kSyntaxErr);
+        return nullopt;
+      }
+
+      uint64_t num_fields = args.size();
+      if (!absl::SimpleAtoi(ArgS(args, i + 1), &num_fields)) {
+        (*cntx)->SendError(kSyntaxErr);
+        return nullopt;
+      }
+
+      i += 1;
+
+      alias_list = SearchParams::FieldAliasList{};
+      while (alias_list->size() < num_fields) {
+        if (i++ >= args.size()) {
```

Comment:
You return the old one here, for example:

`args.size() == 10` 
`i = 9;`
`i++  == 10` but returns `9`

Check is `if(9 >= 10)` => is `false` 

Next line explodes:
`string_view ident = ArgS(args, 10);`

Reply:
What about this?

Reply:
That's true, will push a fix